runtime/doc/develop.txt

```diff
@@ -654,5 +654,21 @@ External UIs are expected to implement these common features:
 - Support the text decorations/attributes given by |ui-event-hl_attr_define|.
   The "url" attr should be presented as a clickable hyperlink.
 
+Adding Features to core vs plugins ~
+
+Many features are better suited as plugins rather than core features. There
+will never be a surefire way to determine whether a feature should be added to
+the editor or if it should be a plugin, but here are some signs. A feature is
+more likely to be suited as a plugin if...
+
+- it sucks
+- can easily be achieved without changes to neovim editor
```

Comment:
I guess the "easily" is a bit of a weasel word, people have different gauges on what they constitute as easy. Also, "easy" probably should relate to how easy it is compared to the value of the feature, rather than how easy it is in a vacuum.